client/implementation/src/main/java/io/smallrye/graphql/client/impl/GraphQLClientsConfiguration.java

```diff
@@ -55,40 +60,32 @@ public GraphQLClientsConfiguration() {
         }
         for (String clientName : detectedClientNames) {
             GraphQLClientConfiguration configuration = new GraphQLClientConfiguration();
-            configuration.setUrl(mpConfig.getValue(clientName + "/mp-graphql/url", String.class));
-            configuration.setHeaders(getConfigurationValueMap(clientName, "header", mpConfig));
+            mpConfig.getOptionalValue(clientName + "/mp-graphql/url", String.class)
+                    .ifPresent(configuration::setUrl);
+            configuration.setHeaders(getConfiguredHeaders(clientName, mpConfig));
             clients.put(clientName, configuration);
         }
     }
 
     /**
-     * Scan the passed Java interfaces for `@GraphQLClientApi` annotations and create and register
-     * client configuration objects for them. This needs to be called by the runtime some time during initialization,
+     * Scan the passed Java interface for `@GraphQLClientApi` annotations and create and register
+     * client configuration objects for it. This needs to be called by the runtime some time during initialization,
      * before clients are actually created.
      */
-    public void addTypesafeClientApis(List<Class<?>> apis) {
```

Comment:
We have to keep this method. Quarkus calls it. I can't compile Quarkus without it
```
[ERROR] /home/jmartisk/Workspace/quarkus/extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/SmallRyeGraphQLClientRecorder.java:[34,19] cannot find symbol
[ERROR]   symbol:   method addTypesafeClientApis(java.util.List<java.lang.Class<?>>)
[ERROR]   location: variable configBean of type io.smallrye.graphql.client.impl.GraphQLClientsConfiguration
```

I would be fine with having both, where `addTypesafeClientApis` calls `initTypesafeClientApi` in a loop

Reply:
done